### PR TITLE
Fix @Default method brigadier tab completion

### PR DESCRIPTION
**Pull request by DeathRealms**
Fixes tab completion for the default method when brigadier api is enabled (#330)

### CODE REVIEW

**Comment by JOO200** [comment]
Maybe extract the duplicated code in a new method?

File: brigadier/src/main/java/co.aikar.commands/ACFBrigadierManager.java [hunk 1]
```diff
@@ -89,6 +89,12 @@
         root = rootBuilder.build();
         boolean isForwardingCommand = rootCommand.getDefCommand() instanceof ForwardingCommand;
 
+        if (rootCommand.getDefaultRegisteredCommand() != null) {
+            CommandNode<S> paramNode = root;
+            CommandParameter[] parameters = rootCommand.getDefaultRegisteredCommand().parameters;
+            registerParameters(parameters, rootCommand, paramNode, suggestionProvider, executor, permCheckerRoot);
```
Comment (JOO200) [inline brigadier/src/main/java/co.aikar.commands/ACFBrigadierManager.java:95]: Why do you pass the rootCommand and permCheckerRoot here? pretty sure it's enough to pass `rootCommand.getDefaultRegisteredCommand()` and `permCheckerSub`.

Your current implementation will provide wrong suggestions for @Default command methods with a @CommandPermission.

File: brigadier/src/main/java/co.aikar.commands/ACFBrigadierManager.java [hunk 2]
```diff
@@ -161,4 +149,31 @@
         return root;
     }
 
+    <C> void registerParameters(CommandParameter[] parameters,
```
Comment (JOO200) [inline brigadier/src/main/java/co.aikar.commands/ACFBrigadierManager.java:152]: Doesn't need to be a generic method when you're only using RegisteredCommand and the permCheckerSub

**Review by JOO200 — CHANGES_REQUESTED** [review]

**Comment by DeathRealms** [comment]
I still need to figure out why when registering the parameters for the default method it prevents sub commands you don't have permission for from being removed from tab completion, but they get removed if you don't have permission for both the default and sub command.

**Comment by chickeneer** [comment]
This change looks reasonable. In regards to the latter comment about 

> when registering the parameters for the default method it prevents sub commands you don't have permission for from being removed from tab completion, but they get removed if you don't have permission for both the default and sub command.

I think it clear you are making this comment in general and shouldn't block merging this.

**Pull request merged**